### PR TITLE
cli_parser: Add class method to make options declaration more readable

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
#1860 
https://github.com/Homebrew/brew/pull/3971#discussion_r176933717

### CODE REVIEW

File: Library/Homebrew/cli_parser.rb [hunk 1]
```diff
@@ -4,6 +4,12 @@
 module Homebrew
   module CLI
     class Parser
+      class << self
+        def parse(&block)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cli_parser.rb]: `def self.parse(&block)` would work, right?

Comment (GauthamGoli) [reply Library/Homebrew/cli_parser.rb]: yeah, but I read that this way is more popular in RoR codebase :grimacing:

Comment (GauthamGoli) [reply Library/Homebrew/cli_parser.rb]: Nevermind. Changing it to `def self.parse(&block)`

**Comment by MikeMcQuaid** [comment]
Thanks again @GauthamGoli!

**Pull request merged**